concrete-commons/src/parameters.rs

```diff
@@ -106,13 +106,36 @@ impl GlweDimension {
         GlweSize(self.0 + 1)
     }
 }
+
 /// The number of coefficients of a polynomial.
 ///
-/// Assuming a polynomial $a_0 + a_1X + /dots + a_nX^N$, this returns $N+1$.
+/// Assuming a polynomial $a_0 + a_1X + /dots + a_NX^N$, this returns $N+1$.
 #[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
 #[cfg_attr(feature = "serde_serialize", derive(Serialize, Deserialize))]
 pub struct PolynomialSize(pub usize);
 
+impl PolynomialSize {
+    /// Returns the associated [`PolynomialSizeLog`].
+    pub fn log2(&self) -> PolynomialSizeLog {
+        PolynomialSizeLog((self.0 as f64).log2().ceil() as usize)
+    }
+}
+
+/// The logarithm of the number of coefficients of a polynomial.
+///
+/// Assuming a polynomial $a_0 + a_1X + /dots + a_{2^n-1}X^{2^n-1}$, this returns $n$.
+/// Panics if the associated [`PolynomialSize`] is not a power of two.
```

Comment:
This comment is not accurate, I've fixed it locally and will push it later on along with the corrections from the reviews (to skip a session of AWS tests).